docs/playbooks.md

```diff
@@ -0,0 +1,185 @@
+
+# native command for production
+$ docker run --rm -it \
+    --env-file env.d/production \
+    arnold \
+    ansible-playbook deploy.yml -e "customer=patient0 env_type=staging"
+```
+
+## `create_project.yml`
+
+This playbook only creates a new OpenShift project name defined with the
+following pattern: `{{ env_type }}-{{ customer }}`
+
+### Usage
+
+```bash
+# development
+$ bin/ansible-playbook create_project.yml -e "customer=patient0 env_type=staging"
+
+# native command for production
+$ docker run --rm -it \
+    --env-file env.d/production \
+    arnold \
+    ansible-playbook create_project.yml -e "customer=patient0 env_type=staging"
+```
+
+## `create_secrets.yml`
+
+The `create_secrets.yml` playbook creates OpenShift secrets that may store
+sensible credentials. It uses vaulted Yaml definition files stored in
+`group_vars/secrets/{{ customer }}/{{ env_type }}/*/credentials.vault.yml` to
+create those secrets. It must be invoked with the `--ask-vault-pass` option so
+that Ansible can decrypt vaulted credentials and push secrets to OpenShift.
+
+### Usage
+
+```bash
+# development
+$ bin/ansible-playbook create_secrets.yml --ask-vault-pass -e "customer=patient0 env_type=staging"
+
+# native command for production
+$ docker run --rm -it \
+    --env-file env.d/production \
+    arnold \
+    ansible-playbook create_secrets.yml --ask-vault-pass -e "customer=patient0 env_type=staging"
+```
+
+## `create_config.yml`
+
+This playbook creates required `ConfigMap` objects for each service. Default
+`ConfigMap`s that will be created are stored in
+`templates/configmap/<< service >>`. Know that these `ConfigMap`s can be
+overridden by adding them to
+`files/configmap/{{ customer }}/{{ env_type }}/<< service >>`.
+
+### Usage
+
+```bash
+# development
+$ bin/ansible-playbook create_config.yml -e "customer=patient0 env_type=staging"
+
+# native command for production
+$ docker run --rm -it \
+    --env-file env.d/production \
+    arnold \
+    ansible-playbook create_config.yml -e "customer=patient0 env_type=staging"
+```
+
+## `create_objects.yml`
+
+As you might expect, this playbook creates all OpenShift objects required for a
+project, _i.e._:
+
+* volumes (_aka_ PVC for Persistant Volume Claim)
+* end points (_aka_ EP)
+* deployment configurations (_aka_ DC)
+* services (_aka_ SVC)
+* jobs
+* routes
+
+All templates for these objects are listed in the `group_vars/all/main.yml` file.
+
+### Usage
+
+```bash
+# development
+$ bin/ansible-playbook create_objects.yml -e "customer=patient0 env_type=staging"
+
+# native command for production
+$ docker run --rm -it \
+    --env-file env.d/production \
+    arnold \
+    ansible-playbook create_objects.yml -e "customer=patient0 env_type=staging"
+```
+
+## `create_acme.yml`
+
+We use this playbook to automatically generate [Let's
+Encrypt](https://letsencrypt.org) SSL certificates for every deployment (if the
+current `env_type` is not `development`).
```

Comment:
The certificates issued are automatically renewed by the service running deployed to OpenShift.